Doc/tools/version_next.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+"""
+Replace `.. versionchanged:: next` lines in docs files by the given version.
+
+Applies to all the VersionChange directives. For deprecated-removed, only
+handle the first argument (deprecation version, not the removal version).
+
+"""
+
+import re
+import sys
+import argparse
+from pathlib import Path
+
+DIRECTIVE_RE = re.compile(
+    r'''
+        (?P<before>
+            \s*\.\.\s+
+            (version(added|changed|removed)|deprecated(-removed)?)
+            \s*::\s*
+        )
+        next
+        (?P<after>
+            .*
+        )
+    ''',
+    re.VERBOSE | re.DOTALL,
+)
+
+basedir = (Path(__file__)
+           .parent  # cpython/Tools/doc
+           .parent  # cpython/Tools
+           .parent  # cpython
+           .resolve()
+           )
+docdir = basedir / 'Doc'
```

Comment:
This file is in ``Doc/tools`` not ``Tools/doc`` (perhaps left over from an earlier draft?)
```suggestion
docdir = (Path(__file__)
           .parent  # cpython/Doc/tools
           .parent  # cpython/Doc
           .resolve()
           )
```